### ISSUE TITLE
Use post method for logout 

### ISSUE THREAD

**Issue by arshdkhn1** [JanitorTechnology/janitor#152]
I think using post request for logout is better practice than using get request. Any image or link having src [try clicking me and then open janitor](https://janitor.technology/logout) will logout the user.

**Comment by jankeromnes** [comment]
I agree that our current logout isn't great from a security stand point. Here are some details about this issue:

- We currently log users out with a link to [janitor.technology/logout](https://janitor.technology/logout) (caution, this will actually log you out). The code for the request handler is in [app.js, lines 141-150](https://github.com/JanitorTechnology/janitor/blob/eef648bc6b3555b5df4dfaaa3e7ea136bd9ed41b/app.js#L141-L150).
- We would like to limit this handler to a specific [HTTP method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), e.g. `DELETE` (Arshad suggested `POST`, which could also work, but I believe that `DELETE` makes more sense and is more secure).

To do this, we could replace the `app.route` handler with a method-specific handler (I'll add an example code soon), and then replace the logout link (`<a>`) with an asynchronous `<form>` element (I'll also add an example for this).

**Comment by jankeromnes** [comment]
To fix this issue, we could replace the [current app.route() logout handler](https://github.com/JanitorTechnology/janitor/blob/eef648bc6b3555b5df4dfaaa3e7ea136bd9ed41b/app.js#L141-L150) with a more specific `app.delete()` handler, like so:

```js
app.delete(/^\/logout\/?/, (request, response) => {
  users.logout(request, error => {
    // ...
  });
});
```

And then, we could replace the [current logout link](https://github.com/JanitorTechnology/janitor/blob/eef648bc6b3555b5df4dfaaa3e7ea136bd9ed41b/templates/header.html#L31) with an asynchronous form, like so:

```html
<form action="/logout/" class="ajax-form has-feedback is-submit" method="delete" refresh-after-success="true">
  <button class="btn btn-link" type="submit">Sign out</button>
</form>
```
